### ISSUE TITLE
Positional arguments should appear before optional arguments in help output (or at least configurably do so)

### ISSUE THREAD

**Issue by Sushisource** [tatsuhiro-t/argparse4j#81]
Currently, the auto-generated help shows positional arguments as coming after all the optional arguments.

I'd argue this is bad in more than just a subjective style sense, and leads users to make mistakes that could be easily avoided, as well as being confusing simply in terms of the ordering of a typical user's thought process.

For example, let's say I have a program foo with help output like this:

```
usage: foo [--multi-optional BAR [BAR ...]] baz
```
If you try to use foo the way the help output tells you to, you'll get this error.

`baz: error: too few arguments`

If help is instead
```
usage: foo baz [--multi-optional BAR [BAR ...]]
```

You'll be lead to do the right thing.

Of course, I could write my own usage line, but with many arguments that sort of defeats the purpose.

Now, I do realize this is what argparse does, and it's the traditional convention, etc, etc. So if you are opposed to making this the default behavior I get that, but having it as an option would be very nice.

**Comment by SSNikolaevich** [comment]
I agree that optional and positional arguments should be grouped. But suggested group order looks unusual for me. IMHO auto-generated help should follow some widely used standard, for example, POSIX or GNU guidelines for command line interfaces.
